contracts/src/registry/UserRegistry.sol

```diff
@@ -1,72 +1,285 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.8.13;
+pragma solidity ^0.8.20;
 
-import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
-import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+import {ERC1155SingletonUpgradeable} from "./ERC1155SingletonUpgradable.sol";
+import {IERC1155Singleton} from "./IERC1155Singleton.sol";
 
 import {IRegistry} from "./IRegistry.sol";
 import {IRegistryDatastore} from "./IRegistryDatastore.sol";
-import {BaseRegistry} from "./BaseRegistry.sol";
-import {NameUtils} from "../utils/NameUtils.sol";
-
-contract UserRegistry is BaseRegistry {
-    uint96 public constant SUBREGISTRY_FLAGS_MASK = 0x1;
-    uint96 public constant SUBREGISTRY_FLAG_LOCKED = 0x1;
 
+/**
+ * @title UserRegistry
+ * @dev Default registry for user names that can be deployed using VerifiableFactory.
+ * This registry handles creation and management of subnames.
+ */
+contract UserRegistry is 
+    Initializable,
+    UUPSUpgradeable, 
+    AccessControlUpgradeable, 
+    ERC1155SingletonUpgradeable,
+    IRegistry // IRegistry extends IERC1155Singleton which has ownerOf
+{
+    // =================== Constants ===================
+    
+    uint96 public constant FLAGS_MASK = 0x7;
+    uint96 public constant FLAG_SUBREGISTRY_LOCKED = 0x1;
+    uint96 public constant FLAG_RESOLVER_LOCKED = 0x2;
+    uint96 public constant FLAG_FLAGS_LOCKED = 0x4;
```

Comment:
I think this is at odds with the work @hiddentao is doing to implement a bitmap-based roles model for permissions (see https://github.com/ensdomains/namechain/pull/19)

Reply:
following the changes, will be updating and adapting when merged